portfolio/src/main/webapp/script.js

```diff
@@ -53,27 +52,60 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-// Creates a map and adds it to the page with markers on compost-dropoff locations.
+// Global variables that apply across functions
+var map, heatmap;
+var heatmapData = [];
+
+// Creates a map and heatmap layer
 function initMap() {
   // Coordinates centers map  around NYC
   var myLatlng = {lat: 40.763232, lng: -73.951047};
   var mapOptions = {
     center: myLatlng, 
-    zoom: 12,
+    zoom: 14,
     mapTypeControl: false 
   };
-  const map = new google.maps.Map(document.getElementById('map'), mapOptions);
+  map = new google.maps.Map(document.getElementById('map'), mapOptions);
```

Comment:
Is this variable used anywhere?

Reply:
Yes - whenever anything is added, it adds to the map object, and the heatmap is based upon this 'map' as well.

Reply:
Ahh this is global. Totally forgot. I see now. Thank you.